src/query-builder/delete-query-builder.ts

```diff
@@ -642,6 +645,35 @@ export class DeleteQueryBuilder<DB, TB extends keyof DB, O>
 
     return result as O
   }
+
+  /**
+   * Executes query with `explain` statement before `delete` keyword.
+   *
+   * ```ts
+   * const explained = await db
+   *  .deleteFrom('person')
+   *  .where('id', '=', 123)
+   *  .explain('json')
+   * ```
+   *
+   * ```sql
+   * explain format=json delete from `person` where `id` = ?
+   * ```
+   */
+  async explain<ER extends Record<string, any> = Record<string, any>>(
```

Comment:
I think the default output type needs to be `unknown` right? Or do all dialects return a list of objects in all cases?